process/process_openbsd.go

```diff
@@ -227,7 +227,12 @@ func (p *Process) GroupsWithContext(ctx context.Context) ([]int32, error) {
 		return nil, err
 	}
 
-	return k.Groups, nil
+	groups := make([]int32, k.Ngroups)
+	for i := int16(0); i < k.Ngroups; i++ {
```

Comment:
Same concerns as in https://github.com/shirou/gopsutil/pull/965#discussion_r503489889

Reply:
I've just copied a FreeBSD implementation, TBH, and it looks right to me. As for #913 , the darwin code looks valid too, no idea why there is a indexing error. Using the `len(k.Groups)` won't help since it always has a fixed length (hence the `k.Ngroups`: it indicates the amount of meaningful `k.Groups` elements).